packages/core/src/baseclient.ts

```diff
@@ -100,6 +101,19 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
   public captureException(exception: any, hint?: EventHint, scope?: Scope): string | undefined {
     let eventId: string | undefined = hint && hint.event_id;
 
+    if (this._options.autoSessionTracking && scope) {
+      const requestSession = scope.getRequestSession();
+      // Necessary check to ensure this is code block is executed only within a request
+      if (requestSession.status !== undefined) {
+        const scopeExtras = scope.getExtras();
+        if (scopeExtras.unhandledPromiseRejection || scopeExtras.onUncaughtException) {
+          requestSession.status = RequestSessionStatus.Crashed;
```

Comment:
Not sure about what I'm saying, but maybe it would be cleaner to update the session status where we determine that a crash or error happened, instead of relying on the indirection via `scope.getExtras()`.

For example in https://github.com/getsentry/sentry-javascript/blob/533975107d9978bcae0b78f61ac214dff4e9ad14/packages/node/src/integrations/onunhandledrejection.ts#L58-L60

Reply:
Would it be better? 
It seemed cleaner to me if this setting status of request Session is all done in `captureException`, especially that `onunhandledrejection` and `onuncaughtexception` call `captureException`